test_time_difference_reader.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import torch
+import numpy as np
+from torch.utils.data import DataLoader
+from costar_dataset.block_stacking_reader_torch import CostarBlockStackingDataset
+
+from tqdm import tqdm
+
+if __name__ == '__main__':
+    """ To test block_stacking_reader for feature_modes time_difference_images and cross_modal_embeddings
+    After reading the images and vectors using the dataloader, this program asserts that images and poses are in range. 
+    Args: 
+        videos_path - Path to the dataset. The path should point to the costar_block_stacking_dataset_v0.4 folder level 
+                      which contains the blocks_only and blocks_with_plush_toy folders.  
+        feature_mode - Options include cross_modal_embeddings and time_difference_images. Default is time_difference_images.
+                       time_difference_images - a feature mode where we try to classify time intervals between two frames.
+                       cross_modal_embeddings - is a mode where we try to classify time intervals between a frame and a contiguous M(default 10) vectors 
+                                                that comprises of encoded pose data, 6 joint angles of the UR5, change in joint angle and state of the gripper.
+        batch_size - Defaults to 32
+        visualize - Set true to view frames being read by the reader. Default is True.  
+    Usage: python test_time_difference_reader.py --videos_path ~/.keras/datasets/costar_block_stacking_dataset_v0.4 --feature_mode cross_modal_embeddings
+    """
+    
+    args_parser = argparse.ArgumentParser()
+    args_parser.add_argument('--videos_path', help='Path to dataset', default='~/.keras/datasets/costar_block_stacking_dataset_v0.4')
+    args_parser.add_argument('--feature_mode', help = 'cross_modal_embeddings or time_difference_images',default='time_difference_images')
+    args_parser.add_argument('--batch_size', help='Batch size for training', type=int, default=32)
+    args_parser.add_argument('--visualize', help='To view frames, set true', default=True)
+    args = args_parser.parse_args()
+
+    visualize = args.visualize
+
+    costar_dataset = CostarBlockStackingDataset.from_standard_txt(
+                      root=args.videos_path,
+                      version='v0.4', set_name='blocks_only', subset_name='success_only',
+                      split='train', feature_mode=args.feature_mode, output_shape=(3, 96, 128),
+                      num_images_per_example=200, is_training=False)
+    generator = DataLoader(costar_dataset, args.batch_size, shuffle=False, num_workers=4)
+    print("Length of the dataset: {}. Length of the loader: {}.".format(len(costar_dataset), len(generator)))
+
+    generator_output = iter(generator)
+    print("-------------------op")
+    x1, x2, y = next(generator_output)
+    print("Image 1 shape: ", x1.shape, "  Image 2/Joint shape: ",x2.shape, "  Labels shape: ", y.shape)
+    
+    pb = tqdm(range(len(generator)-1))
+    for i in pb:
```

Comment:
This is a strange loop. Why not something simpler like the following?
```python
for i, (x1, x2, y) in enumerate(generator):
    # rest of code here
```

Reply:
I used the same code as what was in the end of block_stacking_reader_torch.py. 
You're right, I'll make it simpler.

Reply:
oh but do still use tqdm for progress! :-)